### PR TITLE
Feature/improved controllers and tests

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/ToDo/ShowController.js [hunk 1]
```diff
@@ -41,7 +41,9 @@ class ShowController {
             params: { id: todoId }
         } = request;
 
-        const todo = await this.todoRepository.findById(todoId);
+        const todo = await this.todoRepository.findOne({
```
Comment (daniel-pupczyk-prpl) [inline app/controllers/ToDo/ShowController.js]: use findById but add anohter `where` condition, probably you have to change `findById` in AbstractRepo and use `deepMerge`

File: app/controllers/ToDo/ShowController.js [hunk 2]
```diff
@@ -41,7 +41,9 @@ class ShowController {
             params: { id: todoId }
         } = request;
 
-        const todo = await this.todoRepository.findById(todoId);
+        const todo = await this.todoRepository.findById(todoId, {
+            where: { userId: request.loggedUserId }
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/ShowController.js]: you can destructure `loggedUserId` above

File: app/controllers/ToDo/UpdateController.js [hunk 3]
```diff
@@ -49,28 +48,17 @@ class UpdateController {
...
-            const newTodo = await this.todoRepository.create({
-                ...fields,
-                todoId
-            });
-
-            return response.status(StatusCodes.CREATED).send(newTodo);
+            return response.sendStatus(StatusCodes.NOT_FOUND);
         }
 
         if (todo.userId !== request.loggedUserId) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/controllers/ToDo/UpdateController.js]: as above

File: app/repositories/AbstractRepository.js [hunk 4]
```diff
@@ -12,7 +14,9 @@ class AbstractRepository {
     }
 
     findById(id, options = {}) {
-        return this.model.findByPk(id, options);
+        const where = { where: { id } };
+
+        return this.model.findOne(merge(options, where));
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/repositories/AbstractRepository.js]: just `merge(options, { where: { id } })`
without `where` declaration

File: app/repositories/AbstractRepository.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+const merge = require('deepmerge');
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/repositories/AbstractRepository.js]: const deepmerge

**Review by daniel-pupczyk-rmtcfm-com — APPROVED** [review]

**Pull request merged**